tests/tests/kernel/datatypes/ezuser/ezuser_test.php

```diff
@@ -59,9 +59,14 @@ public function testPasswordHashSamePasswordToUser()
         {
             $this->fail( "User {$this->username} is not in database.");
         }
-        // Not used in this test
         $passwordHashMD5Password = $rows[0]['password_hash'];
 
+        // The value that is expected to be saved in the ezuser table with the HashType to md5_password
+        $hashMD5Expected = md5( $this->password );
+
+        // Verify that the 2 password hashes saved above are the same
+        $this->assertEquals( $hashMD5Expected, $passwordHashMD5Password );
```

Comment:
I would kind of expect this to now fail as password in db is blowfish and not md5, rest afterwards I would expect to pass. Strange, no local reproduction of this with mysql?

Reply:
All logins here run with `UpdateHash=true`, so at this point the password in db has been changed to `md5_password`. What I don't understand is why it isn't changed to `md5_user` during the next login, which is causing the failing tests. I also have no idea why https://github.com/ezsystems/ezpublish-legacy/pull/1322/commits/7b0136add5f9a06f6e6f819646238a0d3fc15a59 is what triggers it.

Can't reproduce it locally, no.